### PR TITLE
fix(TDI-42308): fix batchMode in tMarketoOutput

### PR DESCRIPTION
**Pull request by undx**
**What is the current behavior?** (You can also link to an open issue here)

https://jira.talendforge.org/browse/TDI-42308

**What is the new behavior?**

- increase batchSize to API max 
- manage batch mode in tMarketoOutput for other entities

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/10/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/10/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/11/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/11/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aundx%2FTDI-42308_fix_batchMode_tMarketoOutput)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/18/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/18/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/2/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/2/)

File: components/components-marketo/components-marketo-definition/src/main/java/org/talend/components/marketo/tmarketocampaign/TMarketoCampaignProperties.java [hunk 1]
```diff
@@ -209,8 +209,8 @@ public void afterCampaignAction() {
     }
 
     public void afterTriggerCampaignForLeadsInBatch() {
-        if (batchSize.getValue() > 100) {
-            batchSize.setValue(100);
+        if (batchSize.getValue() > 300) {
+            batchSize.setValue(300);
```
Comment (chmyga) [inline components/components-marketo/components-marketo-definition/src/main/java/org/talend/components/marketo/tmarketocampaign/TMarketoCampaignProperties.java]: This value is used all over the code. Why not moving it to a Constant and use it everywhere?

File: components/components-marketo/components-marketo-definition/src/main/java/org/talend/components/marketo/tmarketoinput/TMarketoInputProperties.java [hunk 2]
```diff
@@ -351,7 +350,7 @@ public void setupProperties() {
         super.setupProperties();
 
         //
-        batchSize.setValue(100);
+        batchSize.setValue(300);
```
Comment (chmyga) [inline components/components-marketo/components-marketo-definition/src/main/java/org/talend/components/marketo/tmarketoinput/TMarketoInputProperties.java]: Common public constant can be used.

File: components/components-marketo/components-marketo-definition/src/main/java/org/talend/components/marketo/tmarketooutput/TMarketoOutputProperties.java [hunk 3]
```diff
@@ -146,7 +144,7 @@ public TMarketoOutputProperties(String name) {
     public void setupProperties() {
         super.setupProperties();
 
-        batchSize.setValue(100);
+        batchSize.setValue(300);
```
Comment (chmyga) [inline components/components-marketo/components-marketo-definition/src/main/java/org/talend/components/marketo/tmarketooutput/TMarketoOutputProperties.java]: Constant can be used for this value

File: components/components-marketo/components-marketo-runtime/src/main/java/org/talend/components/marketo/runtime/MarketoSink.java [hunk 4]
```diff
@@ -62,6 +61,16 @@ public ValidationResult validate(RuntimeContainer container) {
...
             }
+            Integer bsize = ((TMarketoOutputProperties) properties).batchSize.getValue();
+            if (bsize == null) {
+                vr.setStatus(Result.ERROR);
+                vr.setMessage(messages.getMessage("error.validation.batchSize.empty"));
+                return vr;
+            }
+            if (bsize < 1 || bsize > 300) {
+                ((TMarketoOutputProperties) properties).batchSize.setValue(300);
+                LOG.info(messages.getMessage("error.validation.batchSize.range", 300));
```
Comment (chmyga) [inline components/components-marketo/components-marketo-runtime/src/main/java/org/talend/components/marketo/runtime/MarketoSink.java]: Constant can be used

File: components/components-marketo/components-marketo-runtime/src/main/java/org/talend/components/marketo/runtime/MarketoSink.java [hunk 5]
```diff
@@ -106,9 +114,9 @@ public ValidationResult validate(RuntimeContainer container) {
                     return vr;
                 }
                 if (p.triggerCampaignForLeadsInBatch.getValue()) {
-                    if (p.batchSize.getValue() < 1 || p.batchSize.getValue() > 100) {
-                        p.batchSize.setValue(100);
-                        LOG.info(messages.getMessage("error.validation.batchSize.range", 100));
+                    if (p.batchSize.getValue() < 1 || p.batchSize.getValue() > 300) {
+                        p.batchSize.setValue(300);
+                        LOG.info(messages.getMessage("error.validation.batchSize.range", 300));
```
Comment (chmyga) [inline components/components-marketo/components-marketo-runtime/src/main/java/org/talend/components/marketo/runtime/MarketoSink.java]: Constant can be used

File: components/components-marketo/components-marketo-runtime/src/main/java/org/talend/components/marketo/runtime/client/MarketoSOAPClient.java [hunk 6]
```diff
@@ -660,7 +660,7 @@ public MarketoRecordResult getLeadActivity(TMarketoInputProperties parameters, S
     public MarketoRecordResult getLeadChanges(TMarketoInputProperties parameters, String offset) {
         Schema schema = parameters.schemaInput.schema.getValue();
         Map<String, String> mappings = parameters.mappingInput.getNameMappingsForMarketo();
-        int bSize = parameters.batchSize.getValue() > 100 ? 100 : parameters.batchSize.getValue();
+        int bSize = parameters.batchSize.getValue() > 300 ? 300 : parameters.batchSize.getValue();
```
Comment (chmyga) [inline components/components-marketo/components-marketo-runtime/src/main/java/org/talend/components/marketo/runtime/client/MarketoSOAPClient.java]: Constant can be used.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/14/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/14/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aundx%2FTDI-42308_fix_batchMode_tMarketoOutput)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/15/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/15/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aundx%2FTDI-42308_fix_batchMode_tMarketoOutput)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/16/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/16/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aundx%2FTDI-42308_fix_batchMode_tMarketoOutput)

**Pull request merged**